tests/test_java_violations_reporter.py

```diff
@@ -276,41 +271,35 @@ def test_quality_pregenerated_report(self):
         # We're not guaranteed that the violations are returned
         # in any particular order.
         actual_violations = quality.violations("path/to/file.java")
-        self.assertEqual(len(actual_violations), len(expected_violations))
+        assert len(actual_violations) == len(expected_violations)
         for expected in expected_violations:
-            self.assertIn(expected, actual_violations)
+            assert expected in actual_violations
 
 
-class FindbugsQualityReporterTest(unittest.TestCase):
-    def setUp(self):
-        _patch_so_all_files_exist()
+class TestFindbugsQualityReporterTest:
+    @pytest.fixture(autouse=True)
+    def setup(self, mocker):
         # Paths generated by git_path are always the given argument
-        _git_path_mock = patch(
+        _git_path_mock = mocker.patch(
             "diff_cover.violationsreporters.java_violations_reporter.GitPathTool"
-        ).start()
+        )
         _git_path_mock.relative_path = lambda path: path
         _git_path_mock.absolute_path = lambda path: path
 
-    def tearDown(self):
-        """
-        Undo all patches.
-        """
-        patch.stopall()
-
     def test_no_such_file(self):
         quality = QualityReporter(FindbugsXmlDriver())
 
         # Expect that we get no results
         result = quality.violations("")
-        self.assertEqual(result, [])
+        assert not result, []
```

Comment:
whats going on with this one? I forget what assert does when you have the comma

Reply:
to be honest no clue, but that should be not like this. good catch